Source/Core/Common/MemoryUtil.h

```diff
@@ -9,15 +9,11 @@
 
 namespace Common
 {
-void* AllocateExecutableMemory(size_t size, bool low = true);
 void* AllocateMemoryPages(size_t size);
 void FreeMemoryPages(void* ptr, size_t size);
 void* AllocateAlignedMemory(size_t size, size_t alignment);
 void FreeAlignedMemory(void* ptr);
-void ReadProtectMemory(void* ptr, size_t size);
-void WriteProtectMemory(void* ptr, size_t size, bool executable = false);
-void UnWriteProtectMemory(void* ptr, size_t size, bool allowExecute = false);
+bool MemProtect(void* ptr, size_t size, bool read, bool write, bool exec);
```

Comment:
Having 3 booleans is completely unreadable when looking at call sites. Make this a set of flags.